// plugin-welcome/src/api/models/EntExpertise.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntExpertiseEdges,
    EntExpertiseEdgesFromJSON,
    EntExpertiseEdgesFromJSONTyped,
    EntExpertiseEdgesToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntExpertise
 */
export interface EntExpertise {
    /**
     * ExpertiseName holds the value of the "ExpertiseName" field.
     * @type {string}
     * @memberof EntExpertise
     */
    expertiseName?: string;
    /**
     * 
     * @type {EntExpertiseEdges}
     * @memberof EntExpertise
     */
    edges?: EntExpertiseEdges;
    /**
     * ID of the ent.
     * @type {number}
     * @memberof EntExpertise
     */
    id?: number;
}

export function EntExpertiseFromJSON(json: any): EntExpertise {
    return EntExpertiseFromJSONTyped(json, false);
}

export function EntExpertiseFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntExpertise {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'expertiseName': !exists(json, 'ExpertiseName') ? undefined : json['ExpertiseName'],
        'edges': !exists(json, 'edges') ? undefined : EntExpertiseEdgesFromJSON(json['edges']),
        'id': !exists(json, 'id') ? undefined : json['id'],
    };
}

export function EntExpertiseToJSON(value?: EntExpertise | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'ExpertiseName': value.expertiseName,
        'edges': EntExpertiseEdgesToJSON(value.edges),
        'id': value.id,
    };
}
